gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -121,40 +125,74 @@ public static void main(String[] argv) throws IOException {
       PathMatcher filePattern;
       if (format.equalsIgnoreCase("LTF")) {
         filePattern = FileSystems.getDefault().getPathMatcher("glob:**.ltf.zip");
+      } else if (format.equalsIgnoreCase("ace")){
+        // Source files (unannotated) are in the .sgm format
+        // Only checks each adj subdirectory to avoid duplicate source files
+        // (as fp1, fp2 and timex2 use the same source files)
+        if (lang.equalsIgnoreCase("english")) {
+          filePattern = FileSystems.getDefault()
+                  .getPathMatcher("glob:**/English/**/adj/*.sgm");
+        } else if (lang.equalsIgnoreCase("chinese")){
+          filePattern = FileSystems.getDefault()
+                  .getPathMatcher("glob:**/Chinese/**/adj/*.sgm");
+        } else if (lang.equalsIgnoreCase("arabic")) {
+          filePattern = FileSystems.getDefault()
+                  .getPathMatcher("glob:**/Arabic/**/adj/*.sgm");
+        } else {
+          throw new RuntimeException("The ACE corpus does not contain files of the " +
+                  "specified language");
+        }
       } else if (!compressed){
         filePattern = FileSystems.getDefault().getPathMatcher("glob:**/data/**/**");
       } else {
         filePattern = FileSystems.getDefault().getPathMatcher("glob:**/data/**/*.gz");
       }
 
 
-      try (Stream<Path> corpusFiles = Files.walk(corpusDirPath)) {
-        //noinspection ResultOfMethodCallIgnored
-        corpusFiles
-            .filter(filePattern::matches)
-            // we use allMatch because the inner code will return a boolean indicating whether to
-            // continue
-            .allMatch(
-                concatenatedFile -> {
-                    try (ArticleSource articleSource = getArticleSource(format, compressed,
-                            concatenatedFile)) {
-                      // we batch the documents in groups of 100 so we can get the efficiency gains
-                      // from batching without making huge requests of unbounded size
-                      final Iterable<List<Article>> batchedArticles = partition(articleSource, BATCH_SIZE);
-
-                      boolean shouldContinue = index(client, batchedArticles, indexName, lang,
-                              fractionDocAllowToFail, sentenceLimit);
-                      if (!shouldContinue) {
-                        log.info(
-                                "Indexing terminated early without error, probably due to the user "
-                                        + "requesting a limit on the number of documents indexed");
-                        return false;
-                      }
-                  } catch (Exception e) {
-                    throw new RuntimeException(e);
-                  }
-                  return true;
-                });
+      if (!format.equalsIgnoreCase("ace")) {
```

Comment:
Is it possible to refactor so that more of the code is shared between the ACE and not-ACE cases? There seems to be a lot of duplication.